### DISCUSSION TITLE
[feature request] shorter command-line options

### DISCUSSION THREAD

**Discussion by rostok** [ImageMagick/ImageMagick#3647] [Development] [answered]
Hello,
what do you think about **alternative** shorter command-line options to speed up the typing? For example we could have option names shortened to minimal unique prefix. For example instead of typing full `-geometry` only `-ge` would be sufficient as there are no more options staring with `-ge`. In this case `-ge`, `-geom` and `-geometry` would be sufficient.

**Comment by snibgo** [comment]
I think it is a very bad idea. (Sorry!)

My main objection: it makes commands unreadable by humans. So, for example, if I show a script that includes "-ge" then new users will have to read the documentation to discover that this is short for "-geometry".

Next: commands can be easily confused, eg "-ga", "-gb", "-ge", "-gr" etc. This makes writing, debugging and understanding unnecessarily complex.

Next: It makes learning the language far more difficult.

Next: if this was based, as you suggest, on the first significant characters, scripts will break if in the future a new keyword that starts with "-ge" is developed.

**Comment by urban-warrior** [comment, marked as answer]
The ImageMagick command-line is a scripting language and a design principle of language is orthogonality.  There should only be one way to do one thing.  Including aliases for command-line options would violate this principle.  In addition, the current command-line options are theoretically self-documenting, e.g. `-resize` is meaningful, whereas `-re` is less so :-).  In the mean-time, you certainly could write a macro processor to support your proposed aliases.

**Reply by rostok** [reply]
Fair enough. Thanks for clearing this up.
